### PR TITLE
[X86][BF16] Add subvec_zero_lowering patterns

### PR DESCRIPTION
**Pull request by phoebewang**

### CODE REVIEW

File: llvm/lib/Target/X86/X86ISelLowering.cpp [hunk 1]
```diff
@@ -3743,7 +3743,7 @@ static SDValue getZeroVector(MVT VT, const X86Subtarget &Subtarget,
   SDValue Vec;
   if (!Subtarget.hasSSE2() && VT.is128BitVector()) {
     Vec = DAG.getConstantFP(+0.0, dl, MVT::v4f32);
-  } else if (VT.isFloatingPoint()) {
+  } else if (VT.isFloatingPoint() && VT.getVectorElementType() != MVT::bf16) {
```
Comment (RKSimon) [inline llvm/lib/Target/X86/X86ISelLowering.cpp:3746]: Why is this necessary? Are we missing bf16 handling somewhere?

Comment (phoebewang) [reply llvm/lib/Target/X86/X86ISelLowering.cpp:3746]: Because the scalar `bf16` is an illegal type even `vXbf16` are. We worked around it in BUILDVECTOR and should try to avoid build zero vector here.

Comment (RKSimon) [reply llvm/lib/Target/X86/X86ISelLowering.cpp:3746]: Could we test for isTypeLegal(VT.getVectorElementType()) instead?

**Review by RKSimon — APPROVED** [review]
LGTM with one (optional) minor

File: llvm/lib/Target/X86/X86ISelLowering.cpp (hunk 1 above)
Comment (phoebewang) [reply llvm/lib/Target/X86/X86ISelLowering.cpp:3746]: Done.
